src/packages/recompose/__tests__/componentFromStream-test.js

```diff
@@ -102,3 +102,22 @@ test('completed props stream should throw an exception', () => {
 
   expect(() => wrapper.unmount()).toThrowError(/no elements in sequence/)
 })
+
+// TODO: test that this feeds into componentDidCatch() once upgraded to React 16
+test('throws stream error in render', async () => {
+  const Throws = componentFromStream(props$ =>
+    props$.map(props => {
+      if (props.throw) {
```

Comment:
Having that react have ability to intercept errors throwed from render may be we should throw error implicitly inside componentsFromProps without providing throw prop to user?

Reply:
The `throw` here is just for testing, the lib consumers won't see it ;)